jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BuildImageTask.java

```diff
@@ -81,20 +86,45 @@ public void buildImage() throws InvalidImageReferenceException {
               .forToNotConfigured(
                   "'jib.to.image'", "build.gradle", "gradle jib --image <your image name>"));
     }
-    RegistryCredentials knownTargetRegistryCredentials = null;
-    Authorization toAuthorization =
-        ConfigurationPropertyValidator.getImageAuth(
+
+    ImageReference targetImage = ImageReference.parse(jibExtension.getTargetImage());
+
+    CredentialRetrieverFactory credentialRetrieverFactory =
+        CredentialRetrieverFactory.forImage(targetImage, gradleJibLogger);
+    Credential toCredential =
+        ConfigurationPropertyValidator.getImageCredential(
             gradleJibLogger,
             "jib.to.auth.username",
             "jib.to.auth.password",
             jibExtension.getTo().getAuth());
-    if (toAuthorization != null) {
-      knownTargetRegistryCredentials = new RegistryCredentials("jib.to.auth", toAuthorization);
+    RegistryCredentials knownTargetRegistryCredentials = null;
+    CredentialRetriever knownCredentialRetriever = null;
+    if (toCredential != null) {
+      knownTargetRegistryCredentials =
+          new RegistryCredentials(
+              "jib.to.auth",
+              Authorizations.withBasicCredentials(
+                  toCredential.getUsername(), toCredential.getPassword()));
+      knownCredentialRetriever = credentialRetrieverFactory.known(toCredential, "jib.to.auth");
     }
+    // Makes credential retriever list.
+    List<CredentialRetriever> credentialRetrievers = new ArrayList<>();
+    String credentialHelperSuffix = jibExtension.getFrom().getCredHelper();
```

Comment:
This should be `jibExtension.getTo().getCredHelper()`, right? Maybe we should add another test that uses different credential helpers for `from` and `to`.

Reply:
I'll be filing a followup PR to refactor this code / remove the duplication once the `setKnownRegistryCredentials` and `setCredHelper` are removed. It will also make this list-making more testable.